components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.scss

```diff
@@ -155,3 +156,16 @@
   max-height: 365px;
   overflow-x: scroll;
 }
+
+// animation
+.pf-clear-selection {
```

Comment:
A bit of pizzazz to hide/display the button--thanks @SEAjamieD !

Reply:
I'm curious how this works?

Reply:
@bcmdarroch 
It's actually a pretty simple little css animation.

1. I added a base class to the "clear selection" button for easier selection --> `.pf-clear-selection`
2. I added a second dynamic class to the html - `active` that only applies when the selection count is not `0`
3. We set the CSS of the base class so that the button is invisible/not clickable by the user, but still exists in the DOM, taking up space

visibility: hidden --> this hides the elements but keeps it in the DOM
opacity: 0 --> this makes the element see through, so that we can fade it in later
transition: .2s ease-in --> shorthand
         - `.2s` the first number is the time it will take for the element to go from state A to state B (in our case.. to applying the styles in `.active`
        - `ease-in` is a predefined cubic-bezier (a timing function) that determines what the transition speed curve 
         looks like.  We actually want to use `ease-in` (or define our own curve) when an element is LEAVING the page
         - we could actually make this more specific by adding `opacity` as a third property on transition, since thats 
          really the only think we need to animate here.... currently the transition is applying to everything... maybe i'll 
          update that.

I'd like to see us design our own cubic-bezier in the future (and i think we will), but these are a solid place to start.
`ease-out` is kind of like....get there super fast but slow down at the towards the end
`ease-in` is kind of like.... start moving slowly and then crank the speed up.

4.  We add the css for when the `active` class gets applied.  When active is added, the element will fade in.
visibility: visible --> makes the element exist to the user again (thats probably poor choice of words on my part...but I imaging you get the gist)
opacity: 1 --> makes the image full opacity...or not see though
transition: .25s ease-out --> similar to above

You might notice that we are using separate timings (.2s and .25s) - not necessarily super noticeable in this application, but Google Design suggests using 195ms when leaving the page, and 220ms when entering the frame.  In general it just kind of provides a more 'human' motion.

I'm realizing now that everything I typed out makes it look more complicated than it is, but definitely encourage you to play with transitions or `@keyframes` to really have some fun.

Reply:
Great details, @SEAjamieD , thanks! But I have to ask: if recommendations are .195 and .220, why .2 and .25 then? 😁

Reply:
😂 fair question.  The guideline is loose - the more important piece is that the outgoing animation should be slightly faster than the incoming animation.  I felt keeping it to 2 digits read more easily.